### PR TITLE
Add experimental support for DoD HPCMP Blueback, DOD HPCMP ParallelWorks AWS, and clean up documentation (maintainers section)

### PR DESCRIPTION
**Pull request by climbfuji**
### Summary

1. Add a preliminary site config for the upcoming DoD HPCMP Blueback system. This config was tested on an early-access test system for the full machine. Support is for oneAPI and GNU only until the actual machine comes online. @areinecke please review this part.
2. Add a preliminary site config for the upcoming DoD HPCMP ParallelWorks cloud system (AWS only). This config was tested on the current Rocky 8 image, which will not be used once the system is inside a secured environment. Support is for GNU only until the correct image is ready.
3. Clean up the maintainers section in the documentation: remove external ecflow and miniconda (both legacy) and other packages that are no longer built externally by the spack-stack team. Delete corresponding modulefile templates. **JCSDA, EPIC, EMC** please review this part

Note that for 2, I need help from JCSDA to fill in the missing pieces for Discover and S4.

### Testing

Compiled the NEPTUNE standalone environment (template `neptune-dev` without `py-xnrl`) on the Blueback test system. This system is air-gapped. Compiled NEPTUNE using this spack-stack environment.

### Applications affected

None

### Systems affected

No existing system affected

### Dependencies

n/a

### Issue(s) addressed

n/a

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [ ] ~~All dependency PRs/issues have been resolved and this PR can be merged.~~

### CODE REVIEW

**Review by RatkoVasic-NOAA — APPROVED** [review]

**Review by areinecke — APPROVED** [review]
Looks okay with me.

File: doc/source/PreConfiguredSites.rst
Comment (eap) [inline doc/source/PreConfiguredSites.rst]: We had a chat this afternoon and Innocent will be maintaining the 1.6 site config for S4 and doing the 1.9 site install. So you can update this and remove JCDDA.

Per your other question in the PR description, you can leave Discover's maintainers as is.

**Review by eap — APPROVED** [review]
Thanks.

File: doc/source/PreConfiguredSites.rst
Comment (climbfuji) [reply doc/source/PreConfiguredSites.rst]: Ok, thanks. I'll make the maintainer SSEC for S4. And clean up the `MaintainersSection.rst` file before I merge.

**Pull request merged**
